shardingsphere-jdbc/shardingsphere-jdbc-core/src/test/java/org/apache/shardingsphere/driver/jdbc/core/statement/ReadwriteSplittingPreparedStatementTest.java

```diff
@@ -92,4 +94,28 @@ public void assertGetGeneratedKeysWithPrimaryKeyIsNull() throws SQLException {
             assertFalse(generatedKeys.next());
         }
     }
+    
+    @Test
+    public void assertGetGeneratedKeysWithPrimaryKeyIsNullInTransactional() throws SQLException {
+        try (
+                Connection connection = getReadwriteSplittingDataSource()
+                        .getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement("INSERT INTO t_config(id, status) VALUES(?, ?);", Statement.RETURN_GENERATED_KEYS)) {
+            connection.setAutoCommit(false);
+            Object lastGeneratedId = null;
+            Object generatedId = null;
+            for (int i = 1; i <= 3; i++) {
+                preparedStatement.setObject(1, null);
+                preparedStatement.setString(2, "OK");
+                preparedStatement.executeUpdate();
+                ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+                assertTrue(generatedKeys.next());
+                generatedId = generatedKeys.getObject(1);
+                assertNotEquals(lastGeneratedId, generatedId);
```

Comment:
Please use assertThat instead of assertNotEquals.

Reply:
ok

Reply:
> Please use assertThat instead of assertNotEquals.

ok